x-pack/plugins/transform/public/app/common/request.ts

```diff
@@ -57,17 +57,45 @@ export function isDefaultQuery(query: PivotQuery): boolean {
   return isSimpleQuery(query) && query.query_string.query === '*';
 }
 
+export function getCombinedRuntimeMappings(
+  indexPattern: IndexPattern | undefined,
+  runtimeMappings?: StepDefineExposedState['runtimeMappings']
+): StepDefineExposedState['runtimeMappings'] | undefined {
+  let combinedRuntimeMappings = {};
+
+  // Use runtime field mappings defined inline from API
+  if (typeof runtimeMappings === 'object') {
+    combinedRuntimeMappings = { ...combinedRuntimeMappings, ...runtimeMappings };
+  }
```

Comment:
worth checking if it's not `null`

Reply:
Updated here https://github.com/elastic/kibana/commit/bc435134b7a10eb60657c077bec7a52659473569